src/lib.rs

```diff
@@ -72,6 +72,9 @@ fn readline_edit(term: &mut Term, raw: &mut RawMode, history: &History, prompt:
                     },
                     instr::Instr::Noop                   => (),
                     instr::Instr::Cancel                 => return Err(Error::EndOfFile),
+                    instr::Instr::Clear                  => {
+                        let _ = raw.write(b"\x1b[H\x1b[2J");
```

Comment:
Since a write failure is unexpected here, I think it would make sense to wrap this into a `try!` so we explicitly fail if writing is possible.